### DISCUSSION TITLE
milvus实践中的一些问题

### DISCUSSION THREAD

**Discussion by tusm123** [milvus-io/milvus#35086] [Q&A and General discussion]
1、我的数据会频繁删除，如何保证我的segement文件的数据量稳定；目前由于删除有些segement的数据量只有个位数，新的一些都是有10w的数据
2、querynode的内存加载了哪些数据呢？
感谢解答！

**Comment by yhmo** [comment]
下次提问请尽量使用英文。

segment的大小和数量是基于运行时的内外行为决定的：
- 客户端插入数据的频率，间隔，数量会影响segment落盘的时机
- 后台的compaction行为也会不断地合并segment或者重写segment

一般情况下，小segment的数量达到3个或以上，才会触发compaction合并的行为，当某个segment里被删除的数据量达到其行数的10%以上，才会触发重写这个segment。所以，如果有两个比较小的segment，行数比较少，也是符合预期的。

在没有打开mmap的情况下，querynode加载segment的各字段的索引数据，非向量字段的原始数据
